app/services/api/v1x0/catalog/soft_delete.rb

```diff
@@ -9,10 +9,12 @@ def initialize(record)
         end
 
         def process
-          @record.discard
+          @record.discard!
           @restore_key = Digest::SHA1.hexdigest(@record.discarded_at.to_s)
 
           self
+        rescue Discard::RecordNotDiscarded
+          raise Discard::RecordNotDiscarded, "#{@record.class} #{@record.id} #{@record.errors[:base].first}"
```

Comment:
If we are raising a new error, better to a new type defined in the exceptions folder. Discord::RecordNotDiscarded is a type from the gem.
Do we properly catch the error in the controller level and translate to a desired http status code?

Reply:
Yes, `Discord::RecordNotDiscarded` is catched properly in 
https://github.com/RedHatInsights/catalog-api/blob/master/config/initializers/custom_exception_mappings.rb#L21